src/snmalloc/global/libc.h

```diff
@@ -71,11 +70,11 @@ namespace snmalloc::libc
         SNMALLOC_ASSUME(ptr != nullptr);
         ::memcpy(p, ptr, sz);
       }
-      a.dealloc(ptr);
+      dealloc(ptr);
     }
     else if (SNMALLOC_LIKELY(size == 0))
     {
-      a.dealloc(ptr);
+      dealloc(ptr);
```

Comment:
Is the inliner sufficiently clever &| informed to keep this to a single TLS lookup?

Reply:
Good point, I should check that.  I normally only look at malloc and free, and not the disasm for realloc, which generally is not that nice.

Reply:
It accesses the TLS a few times, but there is register pressure as the slow paths cause calls to occur, so it is not clear the code generated is any worse.  The perf in mimalloc-bench is about the same.

Reply:
I'm going to look at optimising realloc after this PR.  So going to leave this for now.